File: src/main/java/fi/quanfoxes/types/Ushort.java
package fi.quanfoxes.types;

import fi.quanfoxes.lexer.NumberType;

public class Ushort extends Number {
    private static final int BYTES = 2;

    public Ushort() {
        super(NumberType.UINT16, 16, "ushort");
    }

    @Override
    public int getSize() {
        return BYTES;
    }
}

File: Zigzag/Types/Ushort.cs
public class Ushort : Number
{
	private const int BYTES = 2;

	public Ushort() : base(NumberType.UINT16, 16, "ushort") { }

	public override int GetReferenceSize()
	{
		return BYTES;
	}

	public override int GetContentSize()
	{
		return BYTES;
	}
}
